pkg/controllers/nodeclaim/lifecycle/registration.go

```diff
@@ -88,13 +92,35 @@ func (r *Registration) syncNode(ctx context.Context, nodeClaim *v1beta1.NodeClai
 	// Sync all taints inside NodeClaim into the Node taints
 	node.Spec.Taints = scheduling.Taints(node.Spec.Taints).Merge(nodeClaim.Spec.Taints)
 	node.Spec.Taints = scheduling.Taints(node.Spec.Taints).Merge(nodeClaim.Spec.StartupTaints)
+	// Remove karpenter.sh/unregistered taint
+	taints, err := isolateKarpenterStartupTaint(node.Spec.Taints)
+	if err != nil {
+		return fmt.Errorf("missing required startup taint: %w", err)
+	}
 	node.Labels = lo.Assign(node.Labels, nodeClaim.Labels, map[string]string{
 		v1beta1.NodeRegisteredLabelKey: "true",
 	})
+	node.Spec.Taints = taints
 	if !equality.Semantic.DeepEqual(stored, node) {
 		if err := r.kubeClient.Patch(ctx, node, client.StrategicMergeFrom(stored)); err != nil {
 			return fmt.Errorf("syncing node labels, %w", err)
 		}
 	}
 	return nil
 }
+
+func isolateKarpenterStartupTaint(ts []v1.Taint) ([]v1.Taint, error) {
```

Comment:
Is this "return all taints that have the same key"? or is it "return the taint that exactly matches this node"? 

If it's the latter, we should just be able to replace this with a `lo.Find()`, since there shouldn't be any collisions. If it's the former, we should use `lo.Filter()`

Reply:
I also don't think it needs to be an error, right? If the taint was already removed/DNE, we just succeed, so this can just be a boolean check + short circuit

Reply:
Yea, moving towards `lo.Reject()` and then using a short circuit after getting the node post status condition checks.